editor/components/inserter/menu.js

```diff
@@ -72,7 +84,11 @@ class InserterMenu extends wp.element.Component {
 						) )
 					}
 				</div>
+				<label htmlFor={ `editor-inserter__search-${ position }` } className="screen-reader-text">
```

Comment:
Can you clarify whether the following would work just as well to associate the label with the input?

```jsx
<label>
	<span className="screen-reader-text">
		{ wp.i18n.__( 'Search blocks' ) }
	</span>
	<input />
</label>
```

I only raise in an effort to avoid `id`, since it's hard to guarantee that a component will only occur once in a page.

Same applies to the above `aria-labelledby`, but an alternative is less clear to me. In cases where I've required an ID in a React component, it can work to store a static incrementing number that's assigned in mount:

```jsx
class Foo extends Component {
	constructor() {
		super( ...arguments );
		this.instanceId = this.constructor.instances++;
	}

	render() {
		return <span id={ 'foo-' + this.instanceId } />;
	}
}

Foo.instances = 0;
```

But, evidenced by this example, it's a bit cumbersome to manage.